dist2src/core.py

```diff
@@ -611,11 +645,19 @@ def copy_all_sources(self, with_patches: bool = False):
         sg_path = self.source_git_path / "SPECS"
         logger.info(f"Copy all sources from {dg_path} to {sg_path}.")
 
-        sources = self.dist_git_spec.get_sources()[:]
+        sources: List[str] = self.dist_git_spec.get_sources()[:]
+        lookaside_source_paths = [path for sha, path in self.lookaside_sources()]
         if with_patches:
             sources += (x.path for x in self.dist_git_spec.get_patches())
 
         for source in sources:
+            # sources are absolute paths as str, lookaside_paths are relative within the repo
+            relative = Path(source).relative_to(self.dist_git_path)
+            if str(relative) in lookaside_source_paths:
+                logger.debug(
+                    f"Source {source} will be fetched from lookaside cache, skipping."
+                )
+                continue
```

Comment:
Why is this required? Once `sources` is set in `.packit.yaml`, the download links for all source archives needed for a build should be available. Which source-archives wouldn't be included in this list that would be fetched by `get_sources.sh` otherwise?

My expectation was, that once we are able to save URLs to the lookaside cache artefacts, we could remove `copy_all_sources()` entirely.

Reply:
There are sources which are directly committed in dist-git - usually config files for services or for the whole distro, e.g. in systemd:
* https://git.centos.org/rpms/systemd/blob/c8s/f/SOURCES (scroll to the end)
* https://git.centos.org/rpms/systemd/blob/c8s/f/SPECS/systemd.spec#_36

Reply:
Meh, what a mess! Totally forgot about this. Thanks for the explanation!

Does there need to be a check to make sure that only files tracked in dist-git are copied and then commited to source-git?

Reply:
I'd say that's what is happening above.

I'll try to get a list of files in SOURCES/ from git directly and intersect it with the set above and print a warning if something is off and see what our tests would tell.